compiler/symbol.go

```diff
@@ -207,7 +214,16 @@ func (info *functionInfo) parsePragmas(f *ssa.Function) {
 				if len(parts) != 2 {
 					continue
 				}
-				info.linkName = parts[1]
+
+				// Set the wasmimport name and the llvm link name
+				info.importName = parts[1]
+
+				if info.module == "" {
+					info.linkName = info.importName
+				} else {
+					info.linkName = fmt.Sprintf("tinygo_wasm_import_%s_%s", info.module, info.importName)
```

Comment:
You are here introducing an order dependency between `//go:wasm-module` and `//export`. Also, you're setting `info.importName` for all systems, not just WebAssembly?

Instead of this if/else, I would move this if/else right after the for loop (putting `parts[1]` in a temporary variable). Something like this:

```go
if info.module != "" {
    // WebAssembly import
    info.importName = importName
} else {
    info.linkName = importName
}
```

Note, in particular, that this _doesn't_ change the link name. I think that's fine (and perhaps better): a TinyGo function name is unlikely to conflict with a name from another programming language with names such as `main.someExportedName` or `github.com/user/pkg/subpkg.someFunction`.